File: api/src/test/java/net/blueshell/api/factory/dto/request/UserActivationRequestFactory.java
package net.blueshell.api.factory.dto.request;

import lombok.RequiredArgsConstructor;
import net.blueshell.api.dto.recovery.UserActivationRequest;
import net.blueshell.api.factory.dto.BaseDtoFactory;
import org.springframework.stereotype.Component;

/**
 * Factory for UserActivationRequest test instances.
 */
@Component
@RequiredArgsConstructor
public class UserActivationRequestFactory extends BaseDtoFactory<UserActivationRequest> {

    @Override
    public Class<UserActivationRequest> targetType() {
        return UserActivationRequest.class;
    }

    @Override
    public UserActivationRequest createBasic() {
        UserActivationRequest dto = new UserActivationRequest();
        dto.setToken(unique("tok"));
        return dto;
    }
}

File: api/src/test/kotlin/net/blueshell/api/factory/dto/request/UserActivationRequestFactory.kt
package net.blueshell.api.factory.dto.request

import net.blueshell.api.domain.auth.web.dto.recovery.UserActivationRequest
import net.blueshell.api.factory.dto.BaseDtoFactory
import org.springframework.stereotype.Component

/**
 * Factory for UserActivationRequest test instances.
 */
@Component
class UserActivationRequestFactory : BaseDtoFactory<UserActivationRequest>() {

    override fun targetType(): Class<UserActivationRequest> = UserActivationRequest::class.java

    override fun createBasic(): UserActivationRequest {
        val dto = UserActivationRequest()
        dto.token = unique("tok")
        return dto
    }
}
